f5_openstack_agent/lbaasv2/drivers/bigip/system_helper.py

```diff
@@ -15,40 +15,38 @@
 
 from oslo_log import log as logging
 
-from f5_openstack_agent.lbaasv2.drivers.bigip.network_helper import \
-    NetworkHelper
-from f5_openstack_agent.lbaasv2.drivers.bigip.resource_helper \
-    import BigIPResourceHelper
-from f5_openstack_agent.lbaasv2.drivers.bigip.resource_helper \
-    import ResourceType
+from f5_openstack_agent.lbaasv2.drivers.bigip.resource \
+    import Device
+from f5_openstack_agent.lbaasv2.drivers.bigip.resource \
+    import Folder
 
 LOG = logging.getLogger(__name__)
 
 
 class SystemHelper(object):
 
     def __init__(self):
-        self.exempt_folders = ['/', 'Common']
+        pass
 
     def create_folder(self, bigip, folder):
-        f = bigip.tm.sys.folders.folder
-        f.create(**folder)
+        f = Folder()
+        f.create(bigip, folder)
 
     def delete_folder(self, bigip, folder_name):
-        f = bigip.tm.sys.folders.folder
-        if f.exists(name=folder_name):
-            obj = f.load(name=folder_name)
-            obj.delete()
+        f = Folder()
+        f.delete(bigip, name=folder_name)
 
     def folder_exists(self, bigip, folder):
         if folder == 'Common':
             return True
 
-        return bigip.tm.sys.folders.folder.exists(name=folder)
+        f = Folder()
+        return f.exists(bigip, name=folder)
 
     def get_folders(self, bigip):
         f_collection = []
-        folders = bigip.tm.sys.folders.get_collection()
+        f = Folder()
+        folders = f.get_collection(bigip)
```

Comment:
does not have get_collection() method at all.    maybe you are trying to call get_resources ?   also several other similar calls in this file. e.g. d.get_collection(bigip) in #71, #80, etc.   should modify these.

Reply:
Done.